api/app/signals/apps/email_integrations/actions/assigned_action.py

```diff
@@ -0,0 +1,91 @@
+# SPDX-License-Identifier: MPL-2.0
+# Copyright (C) 2022 Delta10 B.V.
+import logging
+from email.utils import formataddr
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.core.mail import send_mail
+from django.template import Context, Template, loader
+
+from signals.apps.email_integrations.actions.abstract import AbstractSystemAction
+from signals.apps.email_integrations.exceptions import URLEncodedCharsFoundInText
+from signals.apps.email_integrations.models import EmailTemplate
+from signals.apps.signals.models import Department
+
+logger = logging.getLogger(__name__)
+
+User = get_user_model()
+
+
+class AssignedAction(AbstractSystemAction):
```

Comment:
This email action is missing its own specific rule, something like `AssignedRule`. This rule should contain all checks that needs to be valid before this email action can be triggered. Now a couple of these rules are scattered through the code. For example the checks if the previous assigned departments are the same as the currently assigned departments.

Please create a `AssignedRule` and add it to this action. Example:

```
class AssignedRule(AbstractRule):
   def _validate(self, signal):
       # Code that will validate if the action can trigger can be placed here
       return True

class AssignedAction(AbstractSystemAction):
    ...
    rule = AssignedRule()
    ...
```

Reply:
@vanbuiten and I discussed this yesterday and decided to keep the business logic in tasks.py for now and use the different usages of MailService later as an inspiration for refactoring.